scripts/get_collection_statics.py

```diff
@@ -0,0 +1,242 @@
+        lines = file.readlines()
+
+    # Parse the properties file into a dictionary
+    callno_map = {}
+    general_callno_map = {}
+    for pos in range(3, 445):
+        if '=' in lines[pos]:
+            key, value = lines[pos].strip().split('=', 1)
+            key = key.strip()
+            if len(key) >= 2 and key.isalpha():
+                callno_map[key] = value.strip()
+            else:
+                general_callno_map[key.strip()] = value.strip()
+
+    # Create a columns to store the first letter of the callnoletters
+    df['first_letter'] = df['value'].str[0]
+
+    # Group this letter and map with general_callno_map
+    grouped_df = df.groupby('first_letter')['Count'].sum().reset_index()
+
+    grouped_df['mapped_callnoletters'] = grouped_df['first_letter'].apply(
+        lambda x: general_callno_map.get(x.upper(), x))
+    grouped_df.drop(['first_letter'], axis=1, inplace=True)
+    grouped_df['mapped_callnoletters'] = grouped_df['mapped_callnoletters'].apply(lambda x: get_category_name(x))
+
+    # Create a new column to store the mapped values
+    df.drop(['Percent'], axis=1, inplace=True)
+    df.drop(['first_letter'], axis=1, inplace=True)
+    df.sort_values(by='value', ascending=True, inplace=True)
+    df['value'] = df['value'].str.upper()
+
+    return df, grouped_df
+
+def create_dataframe_from_facets(facet_fields):
+    """
+    Create a DataFrame from Solr facet fields output.
+    :param facet_fields: Dictionary containing facet fields and their counts.
+    :return: DataFrame with facet fields and counts.
+    """
+    facet_data = {}
+
+    for field, values in facet_fields.get('facet_fields').items():
+        data = []
+        for i in range(0, len(values), 2):
+            data.append({
+                'value': values[i],
+                'Count': values[i + 1]
+            })
+        facet_data[field] = data
+
+    return facet_data
+
+
+def add_percentage_column(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a column with the percentage of each row in the DataFrame.
+    :param df: DataFrame containing facet fields and counts.
+    :return: DataFrame with an additional percentage column.
+    """
+    total_count = df['Count'].sum()
+    df['Percent'] = ((df['Count'] / total_count) * 100).round(2)
+    return df
+
+
+def sort_dataframe_by_percentage(df):
+    """
+    Sort the DataFrame by the percentage column in descending order.
+    :param df: DataFrame containing facet fields, counts, and percentages.
+    :return: Sorted DataFrame.
+    """
+    df_sorted = df.sort_values(by='Percent', ascending=False)
+    return df_sorted
+
+
+def create_copyrights_status(df: pd.DataFrame) -> pd.DataFrame:
+
+    """Transform the right status into a more readable format
+    :param df: DataFrame containing the right status and counts.
+    :return: DataFrame with the right status in a more readable format.
+    """
+
+    # Ignoring these categories 4,6,16, 8, 26, 27
+    rights_status = {
```

Comment:
@moseshll This might be another kind of use case for the rights API -- mapping different statuses to different classes of access -- this comes up over and over again in different components.